src/lib/containers/entity/annotations/AdditionalPropertiesSchemaField.tsx

```diff
@@ -147,9 +149,11 @@ export function AdditionalPropertiesSchemaField<T>(
     useListState(convertToArray(formData))
 
   // The type determines which widget we show.
-  const [propertyType, setPropertyType] = useState<PropertyType>(
-    guessPropertyType(list),
-  )
+  const [propertyType, setPropertyType] = useState(guessPropertyType(list))
+
+  // If the property type is updated, store it in a new variable where we'll show a warning if data may be lost on coersion
+  const [nextPropertyType, setNextPropertyType] = useState(propertyType)
```

Comment:
When the annotation type is changed, first store it in this temporary/staging variable to see if we need to show a warning